### DISCUSSION TITLE
Github Actions - Uploading an artifact to repository

### DISCUSSION THREAD

**Discussion by drori200** [community/community#25640] [Actions]
<p>Hello there fellow users! I am working on a workflow that setups my required environments (Node and Java) to execute and build my project. I have everything working except one thing: I need to upload a file that is generated by the software directly to the repository and overwriting or replacing the current one if it exists because i have another software that depends on the file. I have tried uploading the file as an artifact and then use the download action but it doesn’t replace the file. Below i have provided the workflow i have, Any help would be much appreciated!.</p>
<pre><code class="lang-auto">on: push

jobs:
  Generate:
    runs-on: windows-2019

    steps:

      - name: Checkout
        uses: actions/checkout@v2

      - name: Setup Java
        uses: actions/setup-java@v2
        with:
          distribution: 'adopt-hotspot'
          java-version: '8'
          java-package: 'jre'

      - name: Setup Node 14.x
        uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          cache: 'npm'
      - run: npm ci

      - name: Remove File # delete the old distribution.json
        uses: JesseTG/rm@v1.0.2
        with:
          path: ${{github.workspace}}\Files\distribution.json

      - name: Generate New distro #generate new one based on pushed commit
        run: npm run start -- g distro

      - uses: actions/upload-artifact@v2
        with:
          name: distribution
          path: ${{github.workspace}}\Files\distribution.json
      
      - name: download artifact
        uses: actions/download-artifact@v2
        with:
          name: distribution
          path: ${{github.workspace}}\Files
</code></pre>

**Comment by kingthorin** [comment, marked as answer]
<p>Just run the necessary git commands to do it.</p>
<p>git add <br>
git commit<br>
git push</p>
<p>Setup a secret that has the necessary permissions</p>

**Comment by drori200** [comment]
<p>That did the trick! I used <a href="https://github.com/stefanzweifel/git-auto-commit-action" rel="noopener nofollow ugc">this action</a> to make it easier on myself and not complicate things too much. Below is the action i am using for commiting. Hope it will be of use for anyone</p>
<pre><code class="lang-auto">- name: Commit changes
        uses: stefanzweifel/git-auto-commit-action@v4
        with:
          commit_message: Automated Change
          branch: master
          commit_options: '--no-verify --signoff'
          commit_author: Author &lt;actions@github.com&gt;
          status_options: '--untracked-files=no'
          skip_dirty_check: true
</code></pre>

**Reply by skramm** [reply]
Hi,
I'm trying to implement this, but it fails because it can't push to the repo
(`remote: Permission to [myrepo] denied to github-actions[bot].`)
@drori200 , would be kind enough to share what secret I have to store on the repo to make this work?

**Reply by drori200** [reply]
Honestly it has been way too long since i made this issue (4 years) and i don't even remember what this is about lol.

**Reply by kingthorin** [reply]
You'd need to create a PAT and store it as a org and repo secret.
You can use a Classic token with the `repo` scope.

**Reply by skramm** [reply]
Thanks for your replies!
Actually, I just found another solution then the one stated on 
https://github.com/stefanzweifel/git-auto-commit-action :
> Set the contents-permission of the default GITHUB_TOKEN to true. (Required to push new commits to the repository)  

Go to repo settings, then (left pane) Actions->General, and get down to "Workflow Permissions" and grant "read and write permissions". The above snippet from @drori200 works then flawlessly.

![ghactions_settings_400](https://github.com/user-attachments/assets/905d7f67-4c74-41be-aa57-e2af94aea9ab)

**Reply by drori200** [reply]
Do note that this snippet was made 4 years ago. You should check if there were any changes / better ways to implement what i did.
